include/boost/math/distributions/dirichlet.hpp

```diff
@@ -223,20 +199,20 @@ class dirichlet_distribution
   } // dirichlet_distribution constructor.
 
   // Get the concentration parameters.
-  RandomAccessContainer &alpha() const { return m_alpha; }
+  const RandomAccessContainer &get_alpha() const { return m_alpha; }
   // Set the concentration parameters.
-  RandomAccessContainer &alpha() { return m_alpha; }
+  RandomAccessContainer &set_alpha() { return m_alpha; }
 
   // Get the order of concentration parameters.
-  decltype(m_alpha.size()) &order() const { return m_alpha.size(); }
+  size_t order() const { return m_alpha.size(); }
```

Comment:
I would still use `decltype(m_alpha.size())` just to get rid of annoying sign compare warnings for RandomAccessContainers with signed indexes.

Reply:
that was throwing error (not in scope).

Reply:
I trust that your demo/test program has try'n'catch blocks so that throw error messages are visible, rather than unhelpfully bombing with abort

Reply:
@pabristow : That only happens on Windows command line; in other shells the programs don't abort and do give helpful error messages.

Reply:
@mrityunjay-tripathi: Sounds good.

Reply:
We have had many complaints about not seeing the error messages, so not knowing what platform was being used, I thought it was worth repeating.  I'm sure Microsoft would say it was a feature, not a bug, which I suppose it is...

Reply:
How about?

```cpp
auto order() const { return m_alpha.size(); }
```

Reply:
That's nice.

Reply:
But there are users  (Poor Peasants?) still in the Dark Ages of C++03.  Do we need to cut them off gratuitously just for an auto?  (or perhaps they are they cut off already?)